doc/api/process.md

```diff
@@ -2009,6 +2009,49 @@ if (process.getgroups) {
 This function is only available on POSIX platforms (i.e. not Windows or
 Android).
 
+## `process.getOptionValue(option)`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `option` {string}
+* Returns: {any}
+
+The `process.getOptionValue()` method returns the parsed value of
+that command line option. This method respects [`NODE_OPTIONS`][] value.
+
+For example, assuming the following script for `process-getoptionvalue.js`:
+
+```mjs
+import process from 'node:process';
+
+console.log('--pending-deprecation:', process.getOptionValue('--pending-deprecation'));
+console.log('--conditions:', process.getOptionValue('--conditions'));
```

Comment:
Consider this non-blocking but an alternative approach to implementing this would be to use constants on a `process.options` object. For instance,

```js
process.options.pendingDeprecation
process.options.conditions
// ... etc
```

I think this would help with @panva's [concern](https://github.com/nodejs/node/pull/46404/files#r1089955089) in relation to experimental status.